src/plots/gl2d/index.js

```diff
@@ -82,6 +81,47 @@ exports.clean = function(newFullData, newFullLayout, oldFullData, oldFullLayout)
             delete oldFullLayout._plots[id];
         }
     }
+
+    //since we use cartesian interactions, do cartesian clean
+    cartesian.clean.apply(this, arguments)
+};
+
+exports.drawFramework = function(gd) {
```

Comment:
Nice. We probably don't all of this or at least we should be able to reuse the `Cartesian.drawFramework` directly as:

```js
exports.drawFramework = Cartesian.drawFramework;
```